core/src/main/scala/kafka/server/BrokerLifecycleManager.scala

```diff
@@ -244,6 +244,19 @@ class BrokerLifecycleManager(
     eventQueue.append(new OfflineDirEvent(directory))
   }
 
+  def handleKraftJBODMetadataVersionUpdate(): Unit = {
+    eventQueue.append(new KraftJBODMetadataVersionUpdateEvent())
+  }
+
+  private class KraftJBODMetadataVersionUpdateEvent extends EventQueue.Event {
+    override def run(): Unit = {
+      if (!isZkBroker) {
+        registered = false
+        scheduleNextCommunicationImmediately()
```

Comment:
I was thinking if it could cause the heartbeat timeout because we won't send heartbeat when registered is false. But the brokerRegistration will also update the lastContactNs in BrokerHeartbeatManager, so we're good!